flutter-for-android.md

```diff
@@ -0,0 +1,1864 @@
+Fragment represents a way to modularize your code, build more 
+sophisticated user interfaces for larger screens, and help scale your 
+application between small and large screens. In Flutter both of these concepts 
+fall under the concept of a Widget.
+
+## How do I listen to Android Activity lifecycle events
+
+In Android you can override methods from the Activity to capture lifecycle 
+methods for the Activity.
+
+In Flutter You can listen to lifecycle events by hooking into the WidgetsBinding 
+observer and listening to the didChangeAppLifecycleState change event. 
+
+The lifecycle events you can observe are 
+
+- resumed - The application is visible and responding to user input. This is 
+  onResume from Android
+- inactive - The application is in an inactive state and is not receiving user 
+  input. This event is unused on Android and only works with iOS.
+- paused - The application is not currently visible to the user, not responding 
+  to user input, and running in the background. This is onPause from Android
+- suspending - The application will be suspended momentarily. This is unused on 
+  iOS
+
+
+
+<!-- skip -->
+{% prettify dart %}
+import 'package:flutter/widgets.dart';
+
+class LifecycleWatcher extends StatefulWidget {
+  @override
+  _LifecycleWatcherState createState() => new _LifecycleWatcherState();
+}
+
+class _LifecycleWatcherState extends State<LifecycleWatcher> with WidgetsBindingObserver {
+  AppLifecycleState _lastLifecyleState;
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addObserver(this);
+  }
+
+  @override
+  void dispose() {
+    WidgetsBinding.instance.removeObserver(this);
+    super.dispose();
+  }
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    setState(() {
+      _lastLifecyleState = state;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (_lastLifecyleState == null)
+      return new Text('This widget has not observed any lifecycle changes.', textDirection: TextDirection.ltr);
+    return new Text('The most recent lifecycle state this widget observed was: $_lastLifecyleState.',
+        textDirection: TextDirection.ltr);
+  }
+}
+
+void main() {
+  runApp(new Center(child: new LifecycleWatcher()));
+}
+{% endprettify %}
+
+# Layouts
+
+## What is the equivalent of a LinearLayout
+
+In Android, a LinearLayout is used to lay your widgets out linearly -horizontally 
+or vertically. In Flutter, you can use the Row widget or Column widget to 
+achieve the same result.
+
+If you notice the two code samples are identical with the exception of the 
+"Row" and "Column" widget. The children are the same and this feature can be 
+exploited to develop rich layouts that can change overtime with the same 
+children.
+
+<!-- skip -->
+{% prettify dart %}
+new Row(
+  mainAxisAlignment: MainAxisAlignment.center,
+  children: <Widget>[
+    new Text(
+      'Row One',
```

Comment:
you can remove the trailing commas on these strings inside the Text constructors. One line per new Text()